src/EdjCase.JsonRpc.Router/BuilderExtensions.cs

```diff
@@ -43,16 +43,23 @@ public static IServiceCollection AddJsonRpc(this IServiceCollection serviceColle
 			{
 				throw new ArgumentNullException(nameof(serviceCollection));
 			}
-
+			if (configuration == null)
+			{
+				configuration = new RpcServerConfiguration();
+			}
 			serviceCollection.AddSingleton(new RpcServicesMarker());
 			serviceCollection
 				.TryAddScoped<IRpcInvoker, DefaultRpcInvoker>();
 			serviceCollection
 				.TryAddScoped<IRpcParser, DefaultRpcParser>();
 			serviceCollection
-				.TryAddScoped<IRpcRequestHandler, RpcRequestHandler>();
-			serviceCollection
-				.TryAddScoped<IStreamCompressor, DefaultStreamCompressor>();
+				.TryAddScoped<IRpcRequestHandler, RpcRequestHandler>();
+			if (configuration.UseCompression)
+				serviceCollection
+					.TryAddScoped<IStreamCompressor, DefaultStreamCompressor>();
+			else
+				serviceCollection

```

Comment:
I dont think that this line is even needed, I think you can just add if useCompression is true. 
Also, minor but if you could also add { } to the if statement, id appreciate it, so it matches the styling

Thanks for the change!